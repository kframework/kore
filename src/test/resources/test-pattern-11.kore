[]

module TEST-PATTERN-11

  axiom{S}
    \mem{S,S}(X:S,X:S) []

endmodule []
