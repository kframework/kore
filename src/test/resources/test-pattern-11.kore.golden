Definition
    { definitionAttributes = Attributes []
    , definitionModules =
        [ Module
            { moduleName = ModuleName "TEST-PATTERN-11"
            , moduleSentences =
                [ MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters =
                        [ UnifiedObject (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                        ]
                    , sentenceAxiomPattern =
                        KorePattern
                            { getKorePattern = CofreeT
                                { runCofreeT = Identity
                                    { runIdentity = UnifiedObject Null :< UnifiedObjectPattern (InPattern In
                                        { inOperandSort =
                                            SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                                        , inResultSort =
                                            SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                                        , inContainedChild =
                                            CofreeT
                                                { runCofreeT = Identity
                                                    { runIdentity = UnifiedObject Null :< UnifiedObjectPattern (VariablePattern Variable
                                                        { variableName = (Id "X" AstLocationNone) :: Id Object
                                                        , variableCounter = Nothing
                                                        , variableSort =
                                                            SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                                                        })
                                                    }
                                                }
                                        , inContainingChild =
                                            CofreeT
                                                { runCofreeT = Identity
                                                    { runIdentity = UnifiedObject Null :< UnifiedObjectPattern (VariablePattern Variable
                                                        { variableName = (Id "X" AstLocationNone) :: Id Object
                                                        , variableCounter = Nothing
                                                        , variableSort =
                                                            SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                                                        })
                                                    }
                                                }
                                        })
                                    }
                                }
                            }
                    , sentenceAxiomAttributes = Attributes []
                    })
                ]
            , moduleAttributes = Attributes []
            }
        ]
    }